python-sdk/src/astro/databases/base.py

```diff
@@ -134,15 +158,15 @@ def run_sql(
                 stacklevel=2,
             )
             sql = kwargs.get("sql_statement")  # type: ignore
-        sql = query_modifier.merge_pre_and_post_queries(sql)
-        # We need to autocommit=True to make sure the query runs. This is done exclusively for SnowflakeDatabase's
-        # truncate method to reflect changes.
-        if isinstance(sql, str):
-            result = self.connection.execute(
-                sqlalchemy.text(sql).execution_options(autocommit=True), parameters
-            )
-        else:
-            result = self.connection.execute(sql, parameters)
+
+        for sql_query in query_modifier.pre_queries:
+            _ = self.run_single_sql_query(sql_query, parameters)
```

Comment:
@tatiana AFAIK query tags can also be used to track the usage of a particular query. Executing them separately will make the tracking difficult right?

Reply:
ref - https://docs.vividcortex.com/general-reference/query-tags/

Reply:
@utkarsharma2 That's a very good question. Our query tags support was designed focused on Snowflake, and the expectation is that the users would give `ALTER SESSION` statements which would be executed before the main `transform` / `run_raw_query`.

If we only had one `ALTER SESSION` statement and ran the query of interest, this approach would be safe since it would be only for that broader query. However, since ATM, the user can potentially give multiple queries to be run before, and after the main statement, all of them would be labelled the same way, which can be prone to errors.

We can also look into how other databases support query tags - I wonder if it already works for Postgres and MySQL since they rely only on comments. I'll try it out.

For now, I suggest we focus on the customer issue of running the feature as it was originally designed - and review the overall approach in a separate PR/ticket, how do you feel about this?

Reply:
I was thinking if it is worth mentioning in the docs string how we run the query and if you have multiple queries then how metrics will be impacted wdyt?

Reply:
@tatiana I understand to the naming is misleading since `QueryModifier` means something else generally. Would it make sense to change it or isolate it Snowflake?

I won't consider this as a blocker since it's a pressing issue.